src/Lifecycle/Hook.php

```diff
@@ -105,10 +108,26 @@ public function handle(Container $container, Event $event, State $state = null):
         }
     }
 
-    public function replay(Container $container, Event $event, State $state = null): void
+    public function replay(Container $container, Event $event, ?State $state, CarbonInterface $now): void
     {
         if ($this->runsInPhase(Phase::Replay)) {
-            $container->call($this->callback, $this->guessParameters($event, $state));
+            $this->withHistoricalNow($now, fn () => $container->call($this->callback, $this->guessParameters($event, $state)));
+        }
+    }
+
+    protected function withHistoricalNow(CarbonInterface $now, Closure $callback)
+    {
+        $immutable_now = CarbonImmutable::getTestNow();
+        $mutable_now = Carbon::getTestNow();
+
+        try {
+            CarbonImmutable::setTestNow($now);
+            Carbon::setTestNow($now);
+
+            $callback();
```

Comment:
Hypothetically, if a callback took a few seconds to process, would the `finally` block still not have the correct time?

Carbons docs say to clear the mock you should just pass in `null`.

> ### Carbon::setTestNow
> To clear the test instance call this method using the default
parameter of null.

Reply:
The `::getTestNow()` calls will return `null` unless another value had been set previously. So in most cases, calling `::setTestNow()` will just reset it to `null`. But in the case that a value had been set in a test, that value will be restored.